### PR TITLE
Upgrade Ruby to 3.3.0

### PR DESCRIPTION
**Pull request by CatalinVoineag**
## Context

We want to use the new chars keyword argument in the SecureRandom#alphanumeric method introduced in Ruby 3.3.

Also updated bundler to 2.5.6

Gemfile
Dockerfile
Github actions

## Guidance to review

If you're using asdf, you will need to update the ruby plugin `asdf plugin update ruby`

### CODE REVIEW

**Comment by CatalinVoineag** [comment]
[Update Ruby to 3.3](https://trello.com/c/QQ1yRDwG/221-update-ruby-to-33)

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-ruby-3.2.2
+ruby-3.3.0
```
Comment (Nitemaeric) [inline .ruby-version:1]: `rbenv` seems to fail to automatically detect the version with this prefix.

```suggestion
3.3.0
```

Running `rbenv install` raised the following error:

```
❯ rbenv install
ruby-build: definition not found: ruby-3.3.0

See all available versions with `rbenv install --list-all'.

If the version you need is missing, try upgrading ruby-build:

  brew upgrade ruby-build
```

File: Gemfile.lock [hunk 2]
```diff
@@ -660,7 +660,7 @@ DEPENDENCIES
   webmock
 
 RUBY VERSION
-   ruby 3.2.2p53
+   ruby 3.3.0p0
 
 BUNDLED WITH
    2.4.22
```
Comment (Nitemaeric) [inline Gemfile.lock:666]: After installing 3.3.0, my local bundler version was also updated. I wonder if we should update the bundler version here as well? `bundle update --bundler`. Likely worth a separate commit, potentially a separate PR.

**Review by Nitemaeric — APPROVED** [review]
Updated `Gemfile.lock` and app work without any issues.

I did run into an issue with `.ruby-version` though. Running `rbenv local 3.3.0`, it updated the `.ruby-version` content to `3.3.0` locally.

File: Gemfile.lock (hunk 2 above)
Comment (CatalinVoineag) [reply Gemfile.lock:666]: I'll do it in a separate commit

**Review by elceebee — APPROVED** [review]

**Pull request merged**